.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: PyPI Upload
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  dist:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: hynek/build-and-inspect-python-package@v2
+
+  publish:
+    needs: [dist]
+    environment: pypi
+    permissions:
+      id-token: write
+      attestations: write
+      contents: read
```

Comment:
This one's default, you don't need it, though it doesn't hurt.

Reply:
Sorry, I just meant the `contents: read` line, not the entire block.

Reply:
I guess it's not obvious that I was commenting on a single line, GitHub shows context when you do one line.

Reply:
Fixed it. Could you please have a look at it again.